src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -1119,6 +1266,11 @@ public void ActivityCurrentNotSetToStopped()
             Assert.Same(started, Activity.Current);
         }
 
+        public void Dispose()
+        {
+            Activity.Current = null;
```

Comment:
Ah, is this how you're cleaning up after the .Starts without .Stops?

Reply:
yes, this cleans up any started activity